src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -923,10 +923,19 @@ protected internal virtual void Renew()
             if (Disposed)
                 ThrowObjectDisposed();
 
-            if (_perCorePathCache.TryPull(out _activeAllocatePathCaches) == false)
-                _activeAllocatePathCaches = new PathCache();
-            if (_perCoreLazyStringValuesList.TryPull(out _allocateStringValues) == false)
-                _allocateStringValues = new FastList<LazyStringValue>(256);
+            if (_activeAllocatePathCaches == null)
+            {
+                if (_perCorePathCache.TryPull(out _activeAllocatePathCaches) == false)
+                    _activeAllocatePathCaches = new PathCache();
+            }
+
+            if (_allocateStringValues == null)
+            {
+                if (_perCoreLazyStringValuesList.TryPull(out _allocateStringValues) == false)
+                    _allocateStringValues = new FastList<LazyStringValue>(256);
+
+                _numberOfAllocatedStringsValues = 0;
+            }
```

Comment:
Same here - don't we need to clean / dispose `_allocateStringValue`. As I understand we just want to reuse `_allocateStringValues ` instance to avoid allocation of new `FastList`. But shouldn't we make sure it doesn't contain any old LSV items?

Reply:
After talking to Pawel now I realized my comment isn't relevant